src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -132,8 +132,29 @@ trait IterableOps[+A] extends Any {
   }
 
   override def toString = s"$className(${mkString(", ")})"
+
+  def sameElements[B >: A](that: Iterable[B]): Boolean = {
+    val these = this.iterator()
+    val those = that.iterator()
+    while (these.hasNext && those.hasNext)
+      if (these.next() != those.next())
+        return false
+    // At that point we know that *at least one* iterator has no next element
+    // If *both* of them have no elements then the collections are the same
+    these.hasNext == those.hasNext
+  }
+
+  /** Method called from equality methods, so that user-defined subclasses can
+    *  refuse to be equal to other collections of the same kind.
+    *  @param   that   The object with which this $coll should be compared
+    *  @return  `true`, if this $coll can possibly equal `that`, `false` otherwise. The test
+    *           takes into consideration only the run-time types of objects but ignores their elements.
+    */
+  def canEqual(that: Any): Boolean = true
```

Comment:
I would also leave `canEqual` (and inheriting from `Equals`) out of `IterableOps` and implement it in the same places where `equals` is overridden.